tools/clang/test/SemaHLSL/v202x/conforming-literals/valid-literals.hlsl

```diff
@@ -0,0 +1,247 @@
+
+_Static_assert(!is_same<__decltype(1), min16uint>::value, "Literals are not min16uint");
+#endif
+
+_Static_assert(!is_same<__decltype(1), int>::value, "Literals are not int");
+_Static_assert(!is_same<__decltype(1), uint>::value, "Literals are not uint");
+_Static_assert(!is_same<__decltype(1), int64_t>::value, "Literals are not int64_t");
+_Static_assert(!is_same<__decltype(1), uint64_t>::value, "Literals are not uint64_t");
+#endif
+
+////////////////////////////////////////////////////////////////////////////////
+// Integer literals With Suffixes
+////////////////////////////////////////////////////////////////////////////////
+
+#if __HLSL_VERSION > 2021
+_Static_assert(is_same<__decltype(1l), int64_t>::value, "1l is int64_t");
+_Static_assert(is_same<__decltype(1ul), uint64_t>::value, "1ul is uint64_t");
+_Static_assert(is_same<__decltype(1lu), uint64_t>::value, "1lu is uint64_t");
+
+// HLSL 2021 does not define a `long long` type, so the suffix should be
+// invalid.
+_Static_assert(is_same<__decltype(1ll), int64_t>::value, "1ll is int64_t");
+_Static_assert(is_same<__decltype(1ull), uint64_t>::value, "1ull is uint64_t");
+_Static_assert(is_same<__decltype(1llu), uint64_t>::value, "1llu is uint64_t");
+
+// Verify that the size of `long long` is the same as the size of `int64_t`.
+_Static_assert(sizeof(__decltype(1ll)) == sizeof(int64_t), "sizeof(1ll) == sizeof(int64_t)");
+_Static_assert(sizeof(__decltype(1llu)) == sizeof(uint64_t), "sizeof(1llu) == sizeof(uint64_t)");
+
+#else
+_Static_assert(is_same<__decltype(1l), int32_t>::value, "1l is int32_t");
+_Static_assert(is_same<__decltype(1ul), uint32_t>::value, "1ul is uint32_t");
+_Static_assert(is_same<__decltype(1lu), uint32_t>::value, "1lu is uint32_t");
+
+_Static_assert(is_same<__decltype(1ll), int64_t>::value, "1ll is int64_t");
+_Static_assert(is_same<__decltype(1ull), uint64_t>::value, "1ull is uint64_t");
+_Static_assert(is_same<__decltype(1llu), uint64_t>::value, "1llu is uint64_t");
+#endif
+
+////////////////////////////////////////////////////////////////////////////////
+// Ternary operators on integer literals
+////////////////////////////////////////////////////////////////////////////////
+
+#if __HLSL_VERSION > 2021
+_Static_assert(is_same<__decltype(B ? 1 : 1), int>::value, "B ? 1 : 1 is int");
+
+_Static_assert(is_same<__decltype(B ? 1l : 1), int64_t>::value, "B ? 1l : 1 is int64_t");
+_Static_assert(is_same<__decltype(B ? 1 : 1l), int64_t>::value, "B ? 1 : 1l is int64_t");
+
+_Static_assert(is_same<__decltype(B ? 1ul : 1), uint64_t>::value, "B ? 1ul : 1 is uint64_t");
+_Static_assert(is_same<__decltype(B ? 1 : 1ul), uint64_t>::value, "B ? 1 : 1ul is uint64_t");
+
+#if !defined(__HLSL_ENABLE_16_BIT)
+_Static_assert(is_same<__decltype(B ? i16 : i16), min16int>::value, "B ? (min16int) : (min16int) is int"); // maybe a bug
+_Static_assert(is_same<__decltype(B ? u16 : u16), min16uint>::value, "B ? (min16uint) : (min16uint) is int"); // maybe a bug
```

Comment:
What's the bug?  Looks like the text description doesn't match the check.

Reply:
Oh! Sorry, that was a note I left for myself that was a bug in my implementation which I fixed. I'll correct the test and remove the comment.